.github/workflows/build-and-test.yml

```diff
@@ -45,6 +45,13 @@ jobs:
         uses: actions/setup-go@v2.1.3
         with:
           go-version: 1.15
+      - name: Restore module cache
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-go-modules
+        with:
+          path: /home/runner/go/pkg/mod
+          key: go-pkg-mod-${{ runner.os }}-${{ hashFiles('./go.mod') }}
```

Comment:
This can be split in a separate PR.

Reply:
👍 sounds good I will remove it from this one

Reply:
Please send an initial PR :)

Reply:
Removed and PR is here: https://github.com/open-telemetry/opentelemetry-collector/pull/2305